### DISCUSSION TITLE
Request for latest version be publish to Maven

### DISCUSSION THREAD

**Discussion by dannyooi** [postgis/postgis-java#103] [Q&A] [answered]
I'm working on a project where I need to use this library to parse `org.locationtech.jts.geom.Geometry` to `JtsGeometry` and vice versa.
However, I face some errors when I try to use some of the methods:
```
java.lang.NoSuchMethodError: org.locationtech.jts.geom.Polygon.getExteriorRing()Lorg/locationtech/jts/geom/LineString;
        at org.postgis.jts.JtsBinaryWriter.estimatePolygon(JtsBinaryWriter.java:357)
        at org.postgis.jts.JtsBinaryWriter.estimateBytes(JtsBinaryWriter.java:296)
        at org.postgis.jts.JtsBinaryWriter.writeBinary(JtsBinaryWriter.java:106)
        at org.postgis.jts.JtsBinaryWriter.writeBinary(JtsBinaryWriter.java:113)
```

I found that the root cause of the issue is because the version of `jts-core` is not up to date. 
As of current version publish in Maven it is using `jts-core:1.16.1`.
Your latest code is using `1.18.1`, but I can't find any latest publish in Maven repository.
May I know is there a plan to officially publish the latest version soon?

**Comment by phillipross** [comment, marked as answer]
"soon" is a relative timeframe 😃

There are some larger breaking changes planned for the driver that have kept us from releasing the development you see against the master branch, but the CI publishes successful builds to the sonatype snapshots repository which anyone can use (with many caveats).

Since it's difficult to estimate a timeframe for stable releases of the new major versions, it's probably better to upgrade JTS in a maintenance branch (such as v2.5.x) and release that.  I'll look into doing this and see if there might be any breaking changes, in which case we'd want to version it as 2.6 instead of 2.5.  I'll try to get to it this week but I can't guarantee.

If this is urgent for you, I'd recommend submitting a pull request with the update and we'll deal with it as a community contribution 😉

**Reply by dannyooi** [reply]
Hi @phillipross,
May I know how to create branch and submit the pull request? Could you guide me on how to do this?

**Reply by yeroc** [reply]
It looks like the only change for the JTS upgrade was bumping the version in the pom.xml file (95fe87a).  I guess we can cherry-pick that one commit once a branch is created?

**Reply by phillipross** [reply]
Yes, I believe it could be as simple as bumping the JTS version in the v2.5-maintenance branch codeline.

Steps would be something like:
1. fork and clone the repo
2. checkout the v2.5-maintenance branch
3. make the changes and build/test
4. commit and push the changes to your fork
5. create the PR off of your fork against the v2.5-maintenance branch of the postgis repo

**Reply by phillipross** [reply]
I wanted to follow-up on this to say that I realized the existing branch was too out of date to work with the CI.  It's been a bit neglected so I'll take care to get the branch synced with the CI.  This will involve cherry-picking various commits from master, so I might as well grab the jts update too.

I've created issue #106 to track this.

**Reply by yeroc** [reply]
Thanks a lot @phillipross !

**Reply by phillipross** [reply]
v2.5.1 is now available in maven central.  The JTS parser has been built against JTS v1.18.1 and Spatial4J v0.8

**Reply by yeroc** [reply]
Awesome!  Noticed the JTS parser appear on Maven Central a couple hours ago.

**Reply by dannyooi** [reply]
Thanks @phillipross
